mlflow/java/client/src/main/java/org/mlflow/tracking/MlflowClient.java

```diff
@@ -636,4 +643,149 @@ private ArtifactRepository getArtifactRepository(String runId) {
+   *
+   *    <pre>
+   *        List<ModelVersionDetailed> detailsList  = getLatestVersions("model",
+   *                                                              Lists.newArrayList<>("Staging"));
+   *
+   *        for (ModelVersionDetailed details : detailsList) {
+   *            System.out.println("Model Name: " + details.getModelVersion()
+   *                                                       .getRegisteredModel()
+   *                                                       .getName());
+   *            System.out.println("Model Version: " + details.getModelVersion().getVersion());
+   *            System.out.println("Current Stage: " + details.getCurrentStage());
+   *        }
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param stages A list of stages
+   * @return The latest model version
+   *         {@link org.mlflow.api.proto.ModelRegistry.ModelVersionDetailed}
+   */
+  public List<ModelVersionDetailed> getLatestVersions(@Nonnull String modelName,
+                                                      @Nonnull Iterable<String> stages) {
+    String json = sendPost("registered-models/get-latest-versions",
+            mapper.makeGetLatestVersion(modelName, stages));
+    GetLatestVersions.Response response =  mapper.toGetLatestVersionsResponse(json);
+    return response.getModelVersionsDetailedList();
+  }
+
+  /**
+   * :: Experimental ::
+   *
+   * This API may change or be removed in a future release without warning.
+   *
+   * Return the model URI containing for the given model version. The model URI can be used
+   * to download the model version artifacts.
+   *
+   *    <pre>
+   *        String modelUri = getModelVersionDownloadUri("model", 0);
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param version The version number of the model
+   * @return The specified model version's URI.
+   */
+  public String getModelVersionDownloadUri(@Nonnull String modelName, long version) {
+    String json = sendPost("model-versions/get-download-uri",
+            mapper.makeGetModelVersionDownloadUri(modelName, version));
+    return mapper.toGetModelVersionDownloadUriResponse(json);
+  }
+
+  /**
+   * :: Experimental ::
+   *
+   * This API may change or be removed in a future release without warning.
+   *
+   * Return a local file or directory containing all artifacts within the given registered model
+   * version. The method will download the model version artifacts to the local file system.
+   *
+   *    <pre>
+   *        File modelVersionFile = downloadModelVersion("model", 0);
+   *    </pre>
+   *
+   * @param modelName The name of the model
+   * @param version The version number of the model
+   * @return A local file or directory ({@ java.io.File}) containing model artifacts
+   */
+  public File downloadModelVersion(@Nonnull String modelName, long version) {
+    String downloadUri = getModelVersionDownloadUri(modelName,
```

Comment:
nit: oneline

Reply:
Sorry, I don't understand. Do you mean like this?

```
public File downloadModelVersion(...)  {

    String downaloadUri = getModelVersionDownloadFromUri(...);

}
```

Reply:
ya that’s all i mean - it looks like the whole function call would fit in one line.